control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -1630,6 +1629,71 @@ func TestReconcileCreateEndpoint(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:          "Terminating Gateway",
+			consulSvcName: "terminating-gateway",
+			k8sObjects: func() []runtime.Object {
+				pod := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "terminating-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							keyManagedBy: managedByValue,
+						},
+						Annotations: map[string]string{
+							annotationGatewayKind: "terminating",
+						},
+					},
+					Status: corev1.PodStatus{
+						PodIP: "1.2.3.4",
+						Conditions: []corev1.PodCondition{
+							{
+								Type:   corev1.PodReady,
+								Status: corev1.ConditionTrue,
+							},
+						},
+					},
+				}
+				endpoint := &corev1.Endpoints{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "service-created",
+						Namespace: "default",
+					},
+					Subsets: []corev1.EndpointSubset{
+						{
+							Addresses: []corev1.EndpointAddress{
+								{
+									IP: "1.2.3.4",
+									TargetRef: &corev1.ObjectReference{
+										Kind:      "Pod",
+										Name:      "terminating-gateway",
+										Namespace: "default",
+									},
+								},
+							},
+						},
+					},
+				}
+				return []runtime.Object{pod, endpoint}
+			},
+			expectedConsulSvcInstances: []*api.CatalogService{
+				{
+					ServiceID:      "terminating-gateway",
+					ServiceName:    "terminating-gateway",
+					ServiceAddress: "1.2.3.4",
+					ServicePort:    8443,
+					ServiceMeta: map[string]string{
+						MetaKeyPodName:         "terminating-gateway",
+						MetaKeyKubeServiceName: "service-created",
+						MetaKeyKubeNS:          "default",
+						MetaKeyManagedBy:       managedByValue,
+					},
+					ServiceTags: []string{},
+				},
+			},
+			expectedProxySvcInstances: []*api.CatalogService{},
+			expectedHealthChecks:      []*api.HealthCheck{},
```

Comment:
Could we add expected health checks to check that we register them correctly?

Reply:
Also need an enterprise test to check that we register gateway in the correct consul namespace